core/src/main/scalajvm/sttp/client3/HttpURLConnectionBackend.scala

```diff
@@ -150,7 +150,13 @@ class HttpURLConnectionBackend private (
     // inspired by: https://github.com/scalaj/scalaj-http/blob/master/src/main/scala/scalaj/http/Http.scala#L542
     val partsWithHeaders = mp.parts.map { p =>
       val contentDisposition = s"${HeaderNames.ContentDisposition}: ${p.contentDispositionHeaderValue}"
-      val otherHeaders = p.headers.map(h => s"${h.name}: ${h.value}")
+      val otherHeaders = p.body match {
+        case StringBody(_, encoding, _) =>
+          withCharset(p.headers.map(h => h.name -> h.value).toMap, encoding).map { case (name, value) =>
```

Comment:
ah, in `Part` you already have a `Seq[Header]` so you can use that to create a `Headers` instance